src/Cpp/1-getting-started/1-3-1-ImageLibrary/ImageLibraryApplication.cpp

```diff
@@ -133,9 +140,161 @@ bool ImageLibraryApplication::Initialize()
 
     _pipelineFactory = std::make_unique<PipelineFactory>(_device);
 
+    FreeImage_Initialise();
     return true;
 }
 
+WRL::ComPtr<ID3D11ShaderResourceView> CreateTextureViewFromDDS(ID3D11Device* device, const std::wstring& pathToDDS)
+{
+    DirectX::TexMetadata metaData = {};
+    DirectX::ScratchImage scratchImage;
+    if (FAILED(DirectX::LoadFromDDSFile(pathToDDS.c_str(), DirectX::DDS_FLAGS_NONE, &metaData, scratchImage)))
+    {
+        std::cout << "DXTEX: Failed to load image\n";
+        return nullptr;
+    }
+
+    WRL::ComPtr<ID3D11Resource> texture = nullptr;
+    if (FAILED(DirectX::CreateTexture(
+            device,
+            scratchImage.GetImages(),
+            scratchImage.GetImageCount(),
+            metaData,
+            &texture)))
+    {
+        std::cout << "DXTEX: Failed to create texture out of image\n";
+        scratchImage.Release();
+        return nullptr;
+    }
+
+    ID3D11ShaderResourceView* srv = nullptr;
+
+    if (FAILED(DirectX::CreateShaderResourceView(
+            device,
+            scratchImage.GetImages(),
+            scratchImage.GetImageCount(),
+            metaData,
+            &srv)))
+    {
+        std::cout << "DXTEX: Failed to create shader resource view out of texture\n";
+        scratchImage.Release();
+        return nullptr;
+    }
+
+    return srv;
+}
+
+
+WRL::ComPtr<ID3D11ShaderResourceView> CreateTextureView(ID3D11Device* device, const std::wstring& pathToTexture)
```

Comment:
we use ansi strings for everything unrelated to files, windows requires wstrings in order to support non-english languages, for example paths would fail if a user ran this from his downloads folder and their username has a chinese character in them.

Reply:
fair